openmc/mgxs/library.py

```diff
@@ -1005,32 +1018,79 @@ def create_mg_library(self, xs_type='macro', xsdata_names=None,
         # Initialize file
         mgxs_file = openmc.MGXSLibrary(self.energy_groups)
 
-        # Create the xsdata object and add it to the mgxs_file
-        for i, domain in enumerate(self.domains):
-            if self.by_nuclide:
-                nuclides = list(domain.get_all_nuclides().keys())
+        if xs_ids is not None:
+            if isinstance(xs_ids, basestring):
+                # If we only have a string lets convert it now to a list
+                # of strings.
+                xs_id_input = xs_ids
+                xs_ids = []
+                for domain in self.domains:
+                    if self.domain_type is 'mesh':
+                        for subdomain in domain.cell_generator():
+                            xs_ids.append(xs_id_input)
+                    else:
+                        xs_ids.append(xs_id_input)
             else:
-                nuclides = ['total']
-            for nuclide in nuclides:
-                # Build & add metadata to XSdata object
-                if xsdata_names is None:
-                    xsdata_name = 'set' + str(i + 1)
+                cv.check_iterable_type('xs_ids', xs_ids, basestring)
+        else:
+            xs_ids = []
+            for domain in self.domains:
+                if self.domain_type is 'mesh':
+                    for subdomain in domain.cell_generator():
```

Comment:
ditto, seems like the `xs_ids` list can be populated much faster as discussed above